cmd/flux/bootstrap_git.go

```diff
@@ -164,7 +164,9 @@ func bootstrapGitCmdRun(cmd *cobra.Command, args []string) error {
 		// Configure repository URL to match auth config for sync.
 		repositoryURL.User = nil
 		repositoryURL.Scheme = "https"
-		repositoryURL.Host = repositoryURL.Hostname()
+		if repositoryURL.Scheme != "https" && repositoryURL.Scheme != "http" {
```

Comment:
Hi, this doesn't look right to me. The line above it sets the scheme and then immediately checks for the same value.
I may be wrong, but the issue seems to be more about not losing the extra port information in the url.
Maybe we can add a functions like:

```go
func hostnameWithPort(u *url.URL) string {
	if u.Port() == "" {
		return u.Hostname()
	}
	return fmt.Sprintf("%s:%s", u.Hostname(), u.Port())
}
```
to ensure that the port is used if it's available and using that to set `repositoryURL.Host` value.

Playground: https://go.dev/play/p/5lhwRw9vFvP

Reply:
Sorry I missed that. The `if` case shall be before setting the scheme to `https`. We just want to make sure we don't keep an `ssh` port when we want to sync using `http`.

Reply:
I feel this statement could really use a descriptive comment to help contributors (and ourselves) understand in the future what we are doing here, what effect it has, and why we are doing it. As I even have a hard time to grasp after 2.5 weeks of vacation.